src/main/scala/com/amazon/deequ/analyzers/QuantileNonSample.scala

```diff
@@ -0,0 +1,307 @@
+  expand()
+
+  // get and set method for the serialize/deserialize functions
+  def getShrinkingFactor: Double = {
+    shrinkingFactor
+  }
+
+  def getSketchSize: Int = {
+    sketchSize
+  }
+
+  def getCurNumOfCompactors: Int = {
+    curNumOfCompactors
+  }
+
+  def getCompactorActualSize: Int = {
+    compactorActualSize
+  }
+
+  def getCompactorTotalSize: Int = {
+    compactorTotalSize
+  }
+
+  def getCompactorItems: Array[Array[T]] = {
+    var ret = ArrayBuffer[Array[T]]()
+    compactors.toArray.foreach {compactor =>
+      ret = ret :+ compactor.buffer.toArray
+    }
+    ret.toArray
+  }
+
+  def getCompactor: ArrayBuffer[NonSampleCompactor[T]] = {
+    compactors
+  }
+
+
+  // TODO: set stricter access modifier here
+  def setCurNumOfCompactors(input : Int): Unit = {
+    curNumOfCompactors = input
+  }
+
+  def setCompactorActualSize(input : Int): Unit = {
+    compactorActualSize = input
+  }
+
+  def setCompactorTotalSize(input : Int): Unit = {
+    compactorTotalSize = input
+  }
+
+  def setCompactor(input: ArrayBuffer[NonSampleCompactor[T]]) : Unit = {
+    compactors = input
+  }
+
+
+
+  // expand a layer of compactor
+  def expand(): Unit = {
+    compactors = compactors :+ new NonSampleCompactor[T]
+    curNumOfCompactors = compactors.length
+    compactorTotalSize = getCompactorCapacityCount
+  }
+
+  private def capacity(height:Int): Int = {
+    2 * (Math.ceil(sketchSize * Math.pow(shrinkingFactor, height) / 2).toInt + 1)
+  }
+
+  /**
+   * update the sketch with a single item
+   *
+   * @param item new item observed by the sketch
+   */
+  def update(item: T): Unit = {
+    compactors(0).buffer = compactors(0).buffer :+ item
+    compactorActualSize = compactorActualSize + 1
+    if (compactorActualSize > compactorTotalSize) {
+      condense()
+    }
+  }
+
+  def condense(): Unit = {
+    breakable {
+      for (height <- compactors.indices) {
+        if (compactors(height).buffer.length >= capacity(height)) {
+          if (height + 1 >= curNumOfCompactors) expand()
+          val output: Array[T] = compactors(height).compact
+          output.foreach(element => compactors(height + 1).buffer = compactors(height + 1).buffer :+ element)
+          compactorActualSize = getCompactorItemsCount
+          break
+        }
+      }
+    }
+  }
+
+  private def compareGreater[T](o1: T, o2: T)(implicit ord: Ordering[T]): Boolean = ord.gt(o1, o2)
+  private def compareSmaller[T](o1: T, o2: T)(implicit ord: Ordering[T]): Boolean = ord.lt(o1,o2)
+
+  /**
+   * Get the map which contains the rank of all items which is currently in the sketch.
+   * @return the sorted Listmap (by key)which contains the rank of all items which is currently in the sketch (RankMap)
+   */
+  def getRankMap(): ListMap[T,Long] = {
+    val sortedOutput = ListMap(output.toSeq.sortBy(_._1):_*)
+    var states = scala.collection.mutable.Map[T,Long]()
+    var runningRank = 0L
+    sortedOutput.foreach{ tuple =>
```

Comment:
please don't use `_1` and `_2`, destructure tuples and use descriptive variable names
please add a space before {

Reply:
Use item to replace _1, and weight to replace _2
Space added.